Xkcd-Comics/ViewModels/ComicsViewModel.swift

```diff
@@ -0,0 +1,138 @@
+            switch result {
+            case .success(let comic):
+                self.handleFetchedComic(comic: comic)
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingPreviousComicAlert = true
+                }
+                print("ComicsVM, fetching previous comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func fetchNextComic() {
+        fetchingService.fetchComic(comicType: .specificComic(comic.num + 1)) { result in
+            switch result {
+            case .success(let comic):
+                self.handleFetchedComic(comic: comic)
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingNextComicAlert = true
+                }
+                print("ComicsVM, fetching next comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func toggleDescription() {
+        isShowingExplanation.toggle()
+    }
+    
+    func toggleInfo() {
+        isShowingInfo.toggle()
+    }
+    
+    func searchComic(searchNum: String) {
+        guard isShowingSearch else {
+            isShowingSearch = true
+            return
+        }
+        // TODO: error warning if the input is not int
+        fetchingService.fetchComic(comicType: .specificComic(Int(searchNum) ?? 1)) { result in
+            switch result {
+            case .success(let comic):
+                DispatchQueue.main.async {
+                    self.comic = comic
+                    _ = self.isSaved(comic: comic)
+                    self.isShowingSearch = false
+                }
+            case .failure(let error):
+                print("ComicsVM, searching next comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func saveAsFavourite(comic: Comic) {
+        guard !isSaved(comic: comic) else {
+            print("already saved")
+            return
+        }
+        DispatchQueue.main.async {
+            _ = self.isSaved(comic: comic)
+        }
+        userDataService.addComicToFavorites(comic: comic)
+    }
+    
+    func getFavoriteComics() {
+        guard let comics = userDataService.favoriteComics() else {
+            return
+        }
+        for comic in comics {
+            print("ComicVM, getting favorite comics: \(comic.title)")
+        }
+    }
+    
+    private func isSaved(comic: Comic) -> Bool {
+        guard let favouriteComics = userDataService.favoriteComics() else {
+            self.isSaved = false
+            return false
+        }
+        if favouriteComics.contains(comic) {
+            self.isSaved = true
+            return true
+        }
+        self.isSaved = false
+        return false
+    }
+    
+    private func handleFetchedComic(comic: Comic) {
+        DispatchQueue.main.async {
+            self.comic = comic
+            _ = self.isSaved(comic: comic)
```

Comment:
Code duplication: Every time `self.comic` is set, you also have a line to update the saved state. It's very likely that future maintainers will forget to add both lines, meaning bugs will happen, and it's also likely that there will be more things to manage in the future, and then we have to go through many places in the code to add that. 

There should be a central place for handling everything that should be handled when a new comic is set.

My preferred concept is to use a property observer

```swift
@Published private(set) var comic: Comic (...) {
    didSet {
        // ... extra handling
    }
}
```

Or you could use a method

```swift
private func setComic(_ comic: Comic) {
    self.comic = comic
    // ... extra handling
}
```

Reply:
This was the only issue that caused me to click "Request changes". Otherwise, I may have clicked just "Comment" instead or maybe even "Approve"